src/api-service/__app__/instance_config/__init__.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import azure.functions as func
+from onefuzztypes.enums import ErrorCode
+from onefuzztypes.models import Error
+from onefuzztypes.requests import InstanceConfigUpdate
+
+from ..onefuzzlib.config import InstanceConfig
+from ..onefuzzlib.endpoint_authorization import call_if_user
+from ..onefuzzlib.events import get_events
+from ..onefuzzlib.request import not_ok, ok, parse_request
+from ..onefuzzlib.user_credentials import parse_jwt_token
+
+
+def get(req: func.HttpRequest) -> func.HttpResponse:
+    return ok(InstanceConfig.fetch())
+
+
+def post(req: func.HttpRequest) -> func.HttpResponse:
+    request = parse_request(InstanceConfigUpdate, req)
+    if isinstance(request, Error):
+        return not_ok(request, context="instance_config_update")
+
+    config = InstanceConfig.fetch()
+
+    user_info = parse_jwt_token(req)
+    if isinstance(user_info, Error):
+        return not_ok(user_info, context="missing user info")
+
+    if config.admins:
```

Comment:
IIUC, we permit an initial deployment scenario where there are _no_ explicit admins, and thus every authenticated CLI user is a de facto admin (the status quo). But now, can't any one of these users submit an instance config update setting (only) themselves as admin?

I'd like to see a much stricter separation between the two scenarios. Ideally, we should only allow the explicitly-allowed admin case, and require that it be seeded by default and/or out of band.